gdnative-core/src/core_types/string.rs

```diff
@@ -260,6 +260,61 @@ impl std::hash::Hash for GodotString {
     }
 }
 
+impl Add<&GodotString> for &GodotString {
+    type Output = GodotString;
+    #[inline]
+    fn add(self, other: &GodotString) -> GodotString {
+        GodotString::from_sys(unsafe { (get_api().godot_string_operator_plus)(&self.0, &other.0) })
+    }
+}
+
+impl<S> Add<S> for &GodotString 
+where 
+    S: AsRef<str>,
+{
+    type Output = GodotString;
+    #[inline]
+    fn add(self, other: S) -> GodotString {
+        self.add(&GodotString::from_str(other))
+    }
+}
+
+impl AddAssign<&GodotString> for GodotString {
+    #[inline]
+    fn add_assign(&mut self, other: &Self) {
+        let _ = std::mem::replace(self, &*self + other);
```

Comment:
You don't really need this `let _ =` here.

Reply:
Should I add a `#[allow(unused_must_use)]` here?

Reply:
I see. Sorry I made a bad suggestion. Maybe use `self = &*self + other` then?

Reply:
All good, I didn't actually realize assigning directly to worked haha